samples/bigquery_create_dataset.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+// [START bigquery_create_dataset]
+async function bigquery_create_dataset(
+  PROJECT_ID = 'YOUR_PROJECT_ID',
+  DATASET_ID = 'YOUR_DATASET_ID'
+) {
+  // Imports the Google Cloud client library
+  const {BigQuery} = require('@google-cloud/bigquery');
+
+  // Creates a client
+  const bigquery = new BigQuery({PROJECT_ID});
+
+  // Creates a new dataset
+  await bigquery
```

Comment:
In an async function, you don't have to do it this way!  Instead:

```js
const [dataset] = await bigquery.createDataset(DATASET_ID);
console.log(`Dataset ${dataset.id} created.`);
```

No need to do any of the error handling stuff there either. Just let it throw!

Reply:
(same feedback for samples below)

Reply:
I'm curious about this because I had some struggles with it while friction logging the samples when they were formatted that way. For example, when I called createDataset with the same DATASET_ID twice in a row, I got an unclear error message within an unhandled promise rejection and I had to manually change the sample to figure out that the dataset had already existed. Is there another way to handle that scenario?

Reply:
Yes!  This line at the bottom:

```js
bigquery_list_datasets(...process.argv.slice(2));
```

Should be ...

```js
bigquery_list_datasets(...process.argv.slice(2)).catch(console.error);
```

For folks who pick this up and just run it, that should take care of things.  For others that are copy-pasting the snippets - your error handling management is something we have to balance.  While doing the try/catch inline makes it easier for people to see the error for a given snippet, it's *never* code you would want to write in production.  That would mask errors, and make the service act unpredictably.  Instead - you would just let it throw, like we're doing. 

If this doesn't make sense - let me know!

Reply:
Makes sense! Thank you!